Code/GraphMol/FileParsers/ProximityBonds.cpp

```diff
@@ -47,6 +47,41 @@ static bool IsBonded(ProximityEntry *p, ProximityEntry *q, unsigned int flags) {
   return dist2 <= radius * radius;
 }
 
+bool SamePDBResidue(AtomPDBResidueInfo *p, AtomPDBResidueInfo *q) {
+    return p->getResidueNumber() == q->getResidueNumber() &&
+           p->getResidueName() == q->getResidueName() &&
+           p->getChainId() == q->getChainId() &&
+           p->getInsertionCode() == q->getInsertionCode();
+}
+
+
+static bool IsBlacklistedAtom(Atom *atom){
+    // blacklist metals, noble gasses and halogens
+    unsigned int elem = atom->getAtomicNum();
+    // make an inverse query (non-metals and metaloids)
+    if((5 <= elem && elem <= 8) ||
+       (14 <= elem && elem <= 16) ||
+       (32 <= elem && elem <= 34) ||
+       (51 <= elem && elem <= 52))
+        return false;
+    else
+        return true;
+}
+
+bool IsBlacklistedPair(Atom *beg_atom, Atom *end_atom) {
+    AtomPDBResidueInfo *beg_info = (AtomPDBResidueInfo *)beg_atom->getMonomerInfo();
+    AtomPDBResidueInfo *end_info = (AtomPDBResidueInfo *)end_atom->getMonomerInfo();

```

Comment:
You should also check that neither `beg_info` nor `end_info` is NULL.
I'm not sure that this is a `PRECONDITION`, since there may be a sensible answer if one is NULL?

Reply:
I have made the validation match the one in `StandardPDBDoubleBond()`